Firestore/core/src/firebase/firestore/local/leveldb_remote_document_cache.cc

```diff
@@ -185,15 +239,26 @@ DocumentMap LevelDbRemoteDocumentCache::GetMatching(
         continue;
       }
 
-      MaybeDocument maybe_doc = DecodeMaybeDocument(it->value(), document_key);
-      if (!query_path.IsPrefixOf(maybe_doc.key().path())) {
+      if (!query_path.IsPrefixOf(document_key.path())) {
         break;
-      } else if (maybe_doc.is_document()) {
-        results = results.insert(maybe_doc.key(), Document(maybe_doc));
       }
+
+      const std::string& contents = it->value();
+      tasks.Execute([this, &results, document_key, contents] {
```

Comment:
Same suggestion re. capturing `document_key` by reference.

Reply:
This turns out not to work. In the first loop, the reference is is bound to an element that does indeed live in the collection, and has a lifetime that is at least as long as the background thread.

In this case, the reference is bound to the one that `LevelDbRemoteDocumentKey` parsed from the leveldb key for the current row. This reference instance is destroyed on the next iteration.